operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java

```diff
@@ -298,37 +297,21 @@ private static void setDefaultAwaitilityTimings() {
   public void cleanup() {
       Log.info("Deleting Keycloak CR");
 
-      // due to https://github.com/operator-framework/java-operator-sdk/issues/2314 we
-      // try to ensure that the operator has processed the delete event from root objects
       // this can be simplified to just the root deletion after we pick up the fix
       // it can be further simplified after https://github.com/fabric8io/kubernetes-client/issues/5838
       // to just a timed foreground deletion
       var roots = List.of(Keycloak.class, KeycloakRealmImport.class);
-      var dependents = List.of(StatefulSet.class, Secret.class, Service.class, Pod.class, Job.class);
-
-      var rootsDeleted = CompletableFuture.allOf(roots.stream()
-              .map(c -> k8sclient.resources(c).informOnCondition(List::isEmpty)).toArray(CompletableFuture[]::new));
-      roots.forEach(c -> k8sclient.resources(c).withGracePeriod(10).delete());
+      roots.forEach(c -> k8sclient.resources(c).delete());
+      // enforce that at least the statefulset / pods are gone
       try {
-          rootsDeleted.get(1, TimeUnit.MINUTES);
+          k8sclient
+                  .apps()
+                  .statefulSets()
+                  .inNamespace(namespace)
+                  .withLabels(Constants.DEFAULT_LABELS).informOnCondition(List::isEmpty).get(20, TimeUnit.SECONDS);
       } catch (Exception e) {
-          // delete event should have arrived quickly because this is a background delete
-          throw new RuntimeException(e);
+          throw KubernetesClientException.launderThrowable(e);
       }
-      dependents.stream().map(c -> k8sclient.resources(c).withLabels(Constants.DEFAULT_LABELS))
-              .forEach(r -> r.withGracePeriod(10).delete());
```

Comment:
Are we sure this PR fixes #34868? We were already killing resources with a grace period here. The forcible killing doesn't happen with these changes as  hinted [here](https://github.com/keycloak/keycloak/issues/34868#issuecomment-2604772007)?

In any case, let's give it a try and we'll see if this pops up again.

Reply:
> We were already killing resources with a grace period here.

The following lines could supercede that with a termination without a grace period.

Reply:
Wouldn't that happen only if the 10 seconds grace period would be passed?

Reply:
No, those aren't blocking calls.

Reply:
Ah, right. We proceed immediately to the forced killiling. Ok, that makes sense, thanks for the context. 👍

Reply:
Still unsure of why this failure only started to happen more recently and seemingly only on openshift.

Reply:
Our OCP env is a bit specific and different from what we run here in minikube. Let's see if this fix helps.